moj-docker-deploy/apps/templates/scripts/run_container.sh

```diff
@@ -0,0 +1,176 @@
+	CLUSTER_NODES_SAFE="[{% for ip, neighbours in ec2_neighbours.items() %} '{{target_container_name}}-{{neighbours['private_dns_name_safe']}}'{% if not loop.last %},{% endif %} {% endfor %}]"
+	HOSTNAME_OPTS=" -h {{target_container_name}}-{{ec2_local_private_dns_name_safe}} "
+	{% endif %}
+
+	{% if 'links' in cdata %}
+	LINK_CONTAINERS="{% for link_config in cdata.get('links') %} --link {{ link_config.get('link') }}:{{ link_config.get('alias', link_config.get('link')) }} {% endfor %} "
+	{% endif %}
+
+	docker run --name="{{ target_container_name }}" {{cdata.get('docker_args', '')}} \
+	    ${LINK_CONTAINERS} \
+	    -e "CLUSTER_NODES=${CLUSTER_NODES}" \
+	    -e "CLUSTER_NODES_SAFE=${CLUSTER_NODES_SAFE}" \
+	                    {{ docker_branchrunner_args }} \
+	    ${HOSTNAME_OPTS} ${HOSTS_OPTS} ${HOSTS_OPTS_SAFE} ${ULIMIT_OPTS} ${ENV_OPTS} ${VOL_OPTS} ${PORT_OPTS} {{container_full_name}}:"$TAG" {{cdata.get('startup_args', '')}}
+}
+
+function container_post_start()
+{
+	echo INFO: post-start: Checking if container {{target_container_name}} is running
+    # Test for success
+    max_retry=10
+    for i in $(seq 1 ${max_retry})
+    do
+        echo INFO: post-start: Introspecting for a running container named {{target_container_name}}
+        found_container=$(docker inspect --format='{% raw %}{{ .State.Running }}{% endraw %}' {{ target_container_name }} 2>/dev/null || true)
+        if [ "${found_container}" != "true" ]; then
+            echo WARNING: post-start: Docker container {{target_container_name}} not started yet, retry ${i}/${max_retry}...
+            sleep 3
+        else
+            echo INFO: post-start: Docker container {{target_container_name}} running, exiting...
+            exit 0
+        fi
+    done
+    echo ERROR: Docker failed to start instance
+    exit 1
+}
+
+function container_stop( )
+{
+	echo  "{{cname}}_container: stopping service"
+	docker stop {{cname}} > /dev/null 2>&1
+	docker rm -f {{cname}} > /dev/null 2>&1
+}
+
+function container_status( )
+{
+	# If the container is non existent
+	exists=$(docker ps -q --filter="name={{cname}}")
+	if [ "x${exists}" == "x" ]; then
+		echo "not running"
+	else
+		# Get the status of the actual container
+		status=$(docker inspect --format={% raw %}'{{.State.Status}}'{% endraw %} {{cname}})
+		echo  "${status}"
+	fi
+}
+
+# Management instructions of the service
+case "$1" in
+	pre_start)
+		echo  "{{cname}}_container: service pre_start"
+		container_pre_start
+		;;
+	start)
+		echo  "{{cname}}_container: service start"
+		container_start
+		;;
+	post_start)
+		echo  "{{cname}}_container: service post_start"
+		container_post_start
+		;;
+	stop)
+		echo  "{{cname}}_container: service stop"
+				container_stop
+		;;
+	reload)
+		echo  "{{cname}}_container: service reload"
+				container_stop
+		sleep  1
+		container_start
+		;;
+	status)
+		echo  "{{cname}}_container: service status"
+		container_status
+		;;
+	*)
+		echo  "Usage: $ 0 {start | stop | reload | status}"
```

Comment:
not sure if it would be worth echoing out what it was called with (is it always just start|stop|reload|status) ? It might help a bit with supportability (Ie: `echo ERROR: main: called with $* which was unrecognised` type error.